Bottom: 75bb07c88c392dd1be2fb5cb73b0fbc614ad1e91
Top:    e821128176a552209cbf910236e1af5f70f169d4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-02 14:01:49 +0000

Refresh of crasy-test

---

diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 446d805..012e3c5 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -138,28 +138,7 @@ draw_arc (ArcType *arc, const BoxType *drawn_area, void *userdata)
 static void
 draw_poly (PolygonType *polygon, const BoxType *drawn_area, void *userdata)
 {
-  if (!polygon->Clipped)
-    return;
-
-  if (gui->graphics->thindraw_pcb_polygon != NULL &&
-      (TEST_FLAG (THINDRAWFLAG, PCB) ||
-       TEST_FLAG (THINDRAWPOLYFLAG, PCB)))
-    gui->graphics->thindraw_pcb_polygon (Output.fgGC, polygon, drawn_area);
-  else
-    gui->graphics->fill_pcb_polygon (Output.fgGC, polygon, drawn_area);
-
-  /* If checking planes, thin-draw any pieces which have been clipped away */
-  if (gui->graphics->thindraw_pcb_polygon != NULL &&
-      TEST_FLAG (CHECKPLANESFLAG, PCB) &&
-      !TEST_FLAG (FULLPOLYFLAG, polygon))
-    {
-      PolygonType poly = *polygon;
-
-      for (poly.Clipped = polygon->Clipped->f;
-           poly.Clipped != polygon->Clipped;
-           poly.Clipped = poly.Clipped->f)
-        gui->graphics->thindraw_pcb_polygon (Output.fgGC, &poly, drawn_area);
-    }
+  gui->graphics->draw_pcb_polygon (Output.fgGC, polygon, drawn_area);
 }
 
 static void
